docs(validation): Improve process for validation from command line

chriswmackey opened the pull request:
The `validate_model.py` script runs both Pydantic validation and does a re-serialization to Python, which will catch errors beyond those currently in Pydantic (like missing objects that are referenced by name).

@mostaphaRoudsari ,
I just wanted to get your thoughts on including this `validate_model.py`  script, which can be called from command line in order to do a full validation of any model JSON file that people pass in. The ReadMe,MD shows how to use it and I think this will be helpful for @MingboPeng to check whether his C# code is producing the desired results.

chriswmackey commented:
Thanks @mostaphaRoudsari .  I will implement a click CLI, then.

chriswmackey commented:
I added a command line interface with click, using the one for honeybee-core as a template.  The only major difference was that I decided to make click a full dependency of the repo since we are already committing ourselves to exclusive Python3 here with Pydantic. I have verified that I can run the `validate-model` command locally.

Please let me know if this looks correct @mostaphaRoudsari and, if so, I'll implement something similar for honeybee.

mostaphaRoudsari commented on dragonfly_schema/cli/__init__.py:
```diff
@@ -0,0 +1,73 @@
+    """Validate a Model JSON file against the Honeybee schema.
+    \b
+    Args:
+        model-json: Full path to a Model JSON file.
```
```suggestion
        model_json: Full path to a Model JSON file.
```

mostaphaRoudsari commented on dragonfly_schema/cli/__init__.py:
```diff
@@ -0,0 +1,73 @@
+_logger = logging.getLogger(__name__)
+
+
+@click.group(help='Commands for validating JSON files.')
```
I assume you will probably have commands more than just the validating commands. For instance you might add add a command to translate the schema to an energy model or a radiance model, etc.

mostaphaRoudsari commented on dragonfly_schema/cli/__init__.py:
```diff
@@ -0,0 +1,73 @@
+_logger = logging.getLogger(__name__)
+
+
+@click.group(help='Commands for validating JSON files.')
```
I assume you will probably have commands more than just the validating commands. For instance you might add add a command to translate the schema to an energy model or a radiance model, etc.

chriswmackey replied:
Oh. Was that going to be the job of `honeybee-schema` CLI? I thought that was more of a job for `honeybee-energy` or honeybee-radiance` CLI. Right now, I wasn't thinking there'd be anything beyond validation and schema documentation in this repo.

mostaphaRoudsari replied:
That's correct but they will extend this command group. This is basically the entry point for all the commands from the extensions too. So I would make it more generic.

mostaphaRoudsari commented on dragonfly_schema/cli/__init__.py:
```diff
@@ -0,0 +1,73 @@
+    click.echo('viiiiiiiiiiiiizzzzzzzzz!')
+
+
+@main.command('validate-model')
```
I suggest to create a group of commands for `validate` and then add model as a sub-command. That will be easier to extend if one wants to add a new command to validate a face or some other object.

mostaphaRoudsari commented on dragonfly_schema/cli/__init__.py:
```diff
@@ -0,0 +1,73 @@
+        Model.parse_file(model_json)
+        click.echo('Pydantic validation passed.')
+        try:
+            import dragonfly.model as df_model
```
Isn't dragonfly a dependency for this module? If it is then you should not need this check.

mostaphaRoudsari replied:
I see that it is not a dependency. I think we should make it a dependency at least for the `cli`.

chriswmackey commented:
I just wanted to say that I addressed all of the comments by adding validation to the dragonfly-core CLI:
https://github.com/ladybug-tools/dragonfly-core/pull/30

So I'm simplifying this PR to just delete the validate.py file.
